### PR TITLE
Fix "Failed to bind to address"

### PR DESCRIPTION
**Pull request by pellared**
## Why

Fixes https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/issues/1364

## What

Use https://andrewlock.net/how-to-automatically-choose-a-free-port-in-asp-net-core/

## Tests

Running both tests 40 times here: https://github.com/pellared/opentelemetry-dotnet-instrumentation/actions/runs/3192866073

### CODE REVIEW

File: test/test-applications/integrations/TestApplication.Logs/TestApplication.Logs.csproj [hunk 1]
```diff
@@ -4,8 +4,4 @@
     <TargetFrameworks>netcoreapp3.1;net6.0</TargetFrameworks>
   </PropertyGroup>
 
-  <ItemGroup>
-    <Compile Include="..\..\..\IntegrationTests\Helpers\TcpPortProvider.cs" />
```
Comment (pjanotti) [inline test/test-applications/integrations/TestApplication.Logs/TestApplication.Logs.csproj:8]: Do we still need this file? It is kind of low risky but it seems better to allocate and then inform like you did in this PR.

**Review by pjanotti — APPROVED** [review]

**Comment by pjanotti** [comment]
Hit #1362, see https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/actions/runs/3192833581/jobs/5210754217#step:4:3448

Kicking it again...

**Comment by pellared** [comment]
Thanks @pjanotti . Actually, I started to think that 3min may be really too short. See https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/1371

File: test/test-applications/integrations/TestApplication.Logs/TestApplication.Logs.csproj (hunk 1 above)
Comment (pellared) [reply test/test-applications/integrations/TestApplication.Logs/TestApplication.Logs.csproj:8]: In general, it is used in all integration tests, here it was included to not copy+paste it to the test app. But the solution I propose in this PR should be more stable.

**Review by rajkumar-rangaraj — APPROVED** [review]

File: test/test-applications/integrations/TestApplication.Logs/TestApplication.Logs.csproj (hunk 1 above)
Comment (pjanotti) [reply test/test-applications/integrations/TestApplication.Logs/TestApplication.Logs.csproj:8]: yep, ideally we get to the point of deleting that source file.

Comment (pellared) [reply test/test-applications/integrations/TestApplication.Logs/TestApplication.Logs.csproj:8]: We would need to use something different than `HttpListener` to do that 😉

Comment (pellared) [reply test/test-applications/integrations/TestApplication.Logs/TestApplication.Logs.csproj:8]: @RassK Is close to achieve it https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/1372

**Pull request merged**